nano/node/blockprocessor.cpp

```diff
@@ -44,11 +44,19 @@ nano::block_processor::block_processor (nano::node & node_a, nano::write_databas
 			this->condition.notify_all ();
 		}
 	};
+	thread = std::thread ([this] () {
+		nano::thread_role::set (nano::thread_role::name::block_processing);
+		this->process_blocks ();
+	});
 }
 
 nano::block_processor::~block_processor ()
 {
 	stop ();
+	if (thread.joinable ())
```

Comment:
Should the thread join be done in the stop function maybe? I am simply wondering and sharing my thoughts.
This is my thinking:
* The join used to be done in the stop function of node class and now it has moved in the destructor.
* The class state_block_signature_verification joins its thread at the stop function too.

Reply:
Putting the thread join inside ::stop makes it block between when the stop is signaled and when the thread actually exits. The idea of putting it in the destructor means we can signal all the different components to stop, and let each thread shut down simultaneously.